controllers/music.go

```diff
@@ -0,0 +1,104 @@
+package controllers
+
+import (
+	"log"
+	"net/http"
+  "encoding/xml"
+	"fmt"
+	"io/ioutil"
+  "html"
+
+	"github.com/UniversityRadioYork/2016-site/structs"
+	"github.com/UniversityRadioYork/2016-site/utils"
+	"github.com/UniversityRadioYork/myradio-go"
+)
+
+// MusicController is the controller for the Music page.
+type MusicController struct {
+	Controller
+}
+
+// Post represents an individual post in the RSS feed
+type Post struct {
+	Title       string `xml:"title"`
+	Link        string `xml:"link"`
+  PubDate     string `xml:"pubDate"`
+  Updated     string `xml:"updated"`
+  Content     string `xml:"encoded"`
+}
+
+// NewMusicController returns a new MusicController with the MyRadio session s
+// and configuration context c.
+func NewMusicController(s *myradio.Session, c *structs.Config) *MusicController {
+	return &MusicController{Controller{session: s, config: c}}
+}
+
+// Get handles the HTTP GET request r for the Music page, writing to w.
+func (sc *MusicController) Get(w http.ResponseWriter, r *http.Request) {
+  // URL of the RSS feed to query
+  url := sc.config.Music.RSSFeed
+
+  isError := false
+
+  // Make HTTP GET request
+  resp, err := http.Get(url)
+  if err != nil {
+    fmt.Printf("Failed to query URL: %v\n", err)
+    isError = true
+  }
+  defer resp.Body.Close()
+
+  // Read the response body
+  xmlData, err := ioutil.ReadAll(resp.Body)
```

Comment:
nit: ioutil is [deprecated](https://pkg.go.dev/io/ioutil) as of Go 1.16, `io.ReadAll` does the same thing